### ISSUE TITLE
Single-line comments in output

### ISSUE THREAD

**Issue by davidchambers** [jashkenas/coffeescript#2365]
[doctest](https://github.com/davidchambers/doctest#readme) brings Python-like doctests to JavaScript <del>(not to CoffeeScript, at this stage)</del> <ins>and CoffeeScript</ins>. For example:

``` javascript
// > Math.product(3, 4, 5)
// 60
// > Math.product(2, "ten")
// NaN
// > Math.product(6)
// undefined
Math.product = function() {
  var idx = arguments.length
  if (idx > 1) {
    var product = arguments[0]
    while (--idx) product *= arguments[idx]
    return product
  }
}
```

For this reason I'd find it useful to have certain single-line CoffeeScript comments appear as single-line JavaScript comments in the compiled output. Perhaps something like…

``` coffee
## > double(6)
## 12
double = (n) -> 2 * n
```

could become…

``` javascript
var double;

// > double(6)
// 12
double = function(n) {
  return 2 * n;
};
```

Perhaps others would find this behaviour useful. Then again, perhaps not (my use case strikes me as an edge case).

**Comment by flying-sheep** [comment]
the label says duplicate, but what’s the number of the open issue about this feature request?

**Comment by flying-sheep** [comment]
can this please be reopened? @jashkenas?

**Comment by GeoffreyBooth** [comment]
Implemented via #4572. Sorry it took a while.
